art/utils/ensemble.py

```diff
@@ -0,0 +1,36 @@
+from art.core import ArtModule
+from art.utils.enums import BATCH, PREDICTION
+
+import torch
+from torch import nn
+
+from typing import List
+from copy import deepcopy
+
+
+class ArtEnsemble(ArtModule):
+    """
+    Base class for ensembles.
+    """
+
+    def __init__(self, models: List[ArtModule]):
+        super().__init__()
+        self.models = nn.ModuleList(models)
+
+    def predict(self, data):
+        predictions = torch.stack([self.predict_on_model_from_dataloader(model, deepcopy(data)) for model in self.models])
```

Comment:
This won't work - predict is expected to return a dictionary to be compatible with compute metrics.

What I'd suggest is to write forward method - which expects a batch and makes prediction - This will be used later by the user. 

In predict use this forward.

Why you do a deepcopy?